Index: eXoApplication/forum/webapp/src/main/webapp/skin/DefaultSkin/webui/Stylesheet.css
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/skin/DefaultSkin/webui/Stylesheet.css	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/skin/DefaultSkin/webui/Stylesheet.css	(working copy)
@@ -673,3 +673,8 @@
 .UIForumPortlet .FORUMImportCategoryForm .FieldLabel {
 	width: 75px;
 }
+
+.UIForumPortlet .DownArrow1Icon .THFixHidden{
+	overflow: hidden;
+	word-wrap: break-word;
+}
\ No newline at end of file
Index: eXoApplication/forum/webapp/src/main/webapp/skin/DefaultSkin/webui/UIForumPortlet/Stylesheet.css
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/skin/DefaultSkin/webui/UIForumPortlet/Stylesheet.css	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/skin/DefaultSkin/webui/UIForumPortlet/Stylesheet.css	(working copy)
@@ -274,8 +274,10 @@
 }
 
 .UIForumPortlet .ContentContainer .UIGrid th {
-	background: url(background/ForumTitleBar1x19.gif) repeat-x;
+	background: #f0f0f0 url(background/ForumTitleBar1x19.gif) repeat-x;
 	border: 1px solid #d2d4dc;
+	overflow: hidden;
+	word-wrap: break-word;
 }
 
 .UIForumPortlet .ContentContainer .UIGrid td {
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategory.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategory.gtmpl	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategory.gtmpl	(working copy)
@@ -240,8 +240,8 @@
 						<th style="width:30px;">&nbsp;</th>
 						<th><%=_ctx.appRes("UICategory.label.Forums");%></th>
 						<th style="width:30%;"><%=_ctx.appRes("UICategory.label.lastpost");%></th>
-						<th style="width:65px;"><%=_ctx.appRes("UICategory.label.thread");%></th>
-						<th style="width:55px;"><%=_ctx.appRes("UICategory.label.post");%></th>
+						<th style="width:75px;"><%=_ctx.appRes("UICategory.label.thread");%></th>
+						<th style="width:65px;"><%=_ctx.appRes("UICategory.label.post");%></th>
 						<% if(isShowMenu){ %>
 						<th style="width:30px;"><input id="UIFORUMCheckAllForum" type="checkbox" class="checkbox" title="Check all" value="4" onClick="eXo.forum.UIForumPortlet.checkAll(this);"/></th>
 						<% } %>
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicContainer.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicContainer.gtmpl	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicContainer.gtmpl	(working copy)
@@ -417,29 +417,37 @@
 								<th style="width:40px;">&nbsp;</th>
 								<th style="width:40px;">&nbsp;</th>
 								<th >
-								<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","name");%>">
-									<a href="javascript:void(0);"><%=_ctx.appRes("UIPageListTopicByUser.label.threadStarter");%></a>
-								</div>
+									<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","name");%>">
+										<a href="javascript:void(0);"><%=_ctx.appRes("UIPageListTopicByUser.label.threadStarter");%></a>
+									</div>
 								</th>
 								<th style="width:40px;"><div class="AttachmentIcon" title="<%=_ctx.appRes("UITopicContainer.label.Attachments");%>"><span></span></div></th>
 								<th style="width:96px;">
-								<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","voteRating");%>">
-									<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.Rating");%></a>
-								</div>
+									<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","voteRating");%>">
+										<div class="THFixHidden" style="width: 80px;">
+											<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.Rating");%></a>
+										</div>
+									</div>
 								</th>
 								<th style="width:130px;">
 									<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","lastPostDate");%>">
-										<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.LastPosts");%></a>
+										<div class="THFixHidden" style="width: 114px;">
+											<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.LastPosts");%></a>
+										</div>
 									</div>
 								</th>
-								<th style="width:60px;">
+								<th style="width:80px;">
 									<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","postCount");%>">
-										<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.Replies");%></a>
+										<div class="THFixHidden" style="width: 64px;">
+											<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.Replies");%></a>
+										</div>
 									</div>
 								</th>
 								<th style="width:60px;">
 									<div class="DownArrow1Icon" onclick="<%=uicomponent.event("SetOrderBy","viewCount");%>">
-										<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.Views");%></a>
+										<div class="THFixHidden" style="width: 44px;">
+											<a href="javascript:void(0);"><%=_ctx.appRes("UITopicContainer.label.Views");%></a>
+										</div>
 									</div>
 								</th>
 								<% if(canEdit) { %>
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIPageListTopicByUser.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIPageListTopicByUser.gtmpl	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIPageListTopicByUser.gtmpl	(working copy)
@@ -44,21 +44,33 @@
 								</th>
 								<th style="width:130px;">
 									<div onclick="<%=uicomponent.event("SetOrderBy","lastPostDate");%>">
-										<a class="DownArrow1Icon"><%=_ctx.appRes("UITopicContainer.label.LastPosts");%></a>
+										<div class="DownArrow1Icon">
+											<div class="THFixHidden" style="width: 114px;">
+												<%=_ctx.appRes("UITopicContainer.label.LastPosts");%>
+											</div>
+										</div>
 									</div>
 								</th>
 								<th style="width:60px;">
 									<div onclick="<%=uicomponent.event("SetOrderBy","postCount");%>">
-										<a class="DownArrow1Icon"><%=_ctx.appRes("UITopicContainer.label.Replies");%></a>
+										<div class="DownArrow1Icon">
+											<div class="THFixHidden" style="width: 44px;">
+												<%=_ctx.appRes("UITopicContainer.label.Replies");%>
+											</div>
+										</div>
 									</div>
 								</th>
 								<th style="width:60px;">
 									<div onclick="<%=uicomponent.event("SetOrderBy","viewCount");%>">
-										<a class="DownArrow1Icon"><%=_ctx.appRes("UITopicContainer.label.Views");%></a>
+										<div class="DownArrow1Icon">
+											<div class="THFixHidden" style="width: 44px;">
+												<%=_ctx.appRes("UITopicContainer.label.Views");%>
+											</div>
+										</div>
 									</div>
 								</th>
 								<%if(isAdmin){ %>
-									<th style="width:40px;"><%=_ctx.appRes("UIPageListTopicByUser.label.Delete");%></th>
+									<th style="width:50px;"><%=_ctx.appRes("UIPageListTopicByUser.label.Delete");%></th>
 								<%}%>
 							</tr>
 						</thead>
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIPageListPostByIP.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIPageListPostByIP.gtmpl	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIPageListPostByIP.gtmpl	(working copy)
@@ -36,7 +36,11 @@
 									</th>
 									<th style="width:130px">
 										<div onclick="<%=uicomponent.event("SetOrderBy","createdDate");%>">
-											<div class="DownArrow1Icon"><%=_ctx.appRes("UIPageListPostByUser.label.CreateDate");%></div>
+											<div class="DownArrow1Icon">
+												<div class="THFixHidden" style="width: 114px;">
+													<%=_ctx.appRes("UIPageListPostByUser.label.CreateDate");%>
+												</div>
+											</div>
 										</div>
 									</th>
 									<th style="width:60px"><%=_ctx.appRes("UIPageListPostByUser.label.Delete");%></th>
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIForumAdministrationForm.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIForumAdministrationForm.gtmpl	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIForumAdministrationForm.gtmpl	(working copy)
@@ -9,7 +9,10 @@
 	List bbcodes = uiform.getListBBcode();
 	String classCssInbox = "";
 %>
-<div class="UIFormInputWithActions UIForumAdministrationForm" style="*width:97%;">	
+<div class="FixIE">
+<div><div class="FixWidth"><span style="display: none;"></span></div></div>
+<div style="_width: expression(this.previousSibling.offsetWidth - 22 + 'px'); _margin: auto;">
+<div class="UIFormInputWithActions UIForumAdministrationForm" style="*width:97%; min-width: 695px;">	
 	<% uiform.begin() %>	
 	<div class="UIFormTabPane">
 		<div class="UITabPane">
@@ -406,4 +409,6 @@
  		</table> 
 	</div>	
 	<%uiform.end()%>
+</div>
+</div>
 </div>
\ No newline at end of file
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategories.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategories.gtmpl	(revision 65579)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategories.gtmpl	(working copy)
@@ -102,8 +102,8 @@
 									<th style="width:30px;">&nbsp;</th>
 									<th><%=_ctx.appRes("UICategory.label.Forums");%></th>
 									<th style="width:30%;"><%=_ctx.appRes("UICategory.label.lastpost");%></th>
-									<th style="width:50px;"><%=_ctx.appRes("UICategory.label.thread");%></th>
-									<th style="width:50px;"><%=_ctx.appRes("UICategory.label.post");%></th>
+									<th style="width:75px;"><%=_ctx.appRes("UICategory.label.thread");%></th>
+									<th style="width:65px;"><%=_ctx.appRes("UICategory.label.post");%></th>
 								</tr>
 							</thead>
 							<tbody>
