Index: eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/UIBreadcumbs.java
===================================================================
--- eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/UIBreadcumbs.java	(revision 79116)
+++ eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/UIBreadcumbs.java	(working copy)
@@ -45,7 +45,7 @@
 	public List<String> pathList_ = new ArrayList<String>();
 	private String currentPath_ = Utils.CATEGORY_HOME ;
 	public static final String FIELD_FAQHOME_BREADCUMBS = "faqHome" ;
-	private static final String QUICK_SEARCH = "QuickSearch";
+	private static final String QUICK_SEARCH = "QuickSearchForm";
 	
 	public UIBreadcumbs()throws Exception {
 		addChild(UIQuickSearch.class, null, QUICK_SEARCH) ;
