Index: eXoApplication/faq/webapp/src/main/webapp/templates/faq/webui/UIBreadcumbs.gtmpl
===================================================================
--- eXoApplication/faq/webapp/src/main/webapp/templates/faq/webui/UIBreadcumbs.gtmpl	(revision 73761)
+++ eXoApplication/faq/webapp/src/main/webapp/templates/faq/webui/UIBreadcumbs.gtmpl	(working copy)
@@ -63,7 +63,7 @@
 						  <% } %>
 						 </div>
 				  </td>
-				  <td style="width:280px;">
+				  <td style="width:310px;">
 						<div class="SearchForm">
 							<%uicomponent.renderChild(uicomponent.QUICK_SEARCH) ;%>
 						</div>
@@ -73,4 +73,4 @@
 		 	</tbody>
 		</table>
 	</div>
-</div>
\ No newline at end of file
+</div>
