Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java	(revision 52058)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java	(working copy)
@@ -1765,7 +1765,7 @@
 	  					} else if((int)c == 62){
 	  						buffer.append("&gt;") ;
 	  					} else if(c == '\''){
-	  						buffer.append("&apos;") ;
+	  						buffer.append("&#39;") ;
 	  					} else if(c == '&' || (int)c == 38){
 	  						buffer.append("&#x26;");
 	  					} else{
