Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java	(revision 68796)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java	(working copy)
@@ -723,6 +723,12 @@
 		}
 		return "";
 	}
+	
+	private String getViewScreenName(String screenName) {
+		if(screenName != null && screenName.trim().length() > 17 && !screenName.contains(" "))
+			screenName = "<spam title=\""+screenName+"\">"+ForumUtils.getSubString(screenName, 16)+"</span>";
+		return screenName;
+	}
 
 	static public class AddPostActionListener extends EventListener<UITopicDetail> {
 		public void execute(Event<UITopicDetail> event) throws Exception {
Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIViewTopic.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIViewTopic.java	(revision 68796)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIViewTopic.java	(working copy)
@@ -27,6 +27,7 @@
 import org.exoplatform.container.PortalContainer;
 import org.exoplatform.download.DownloadService;
 import org.exoplatform.forum.ForumSessionUtils;
+import org.exoplatform.forum.ForumUtils;
 import org.exoplatform.forum.service.ForumAttachment;
 import org.exoplatform.forum.service.ForumService;
 import org.exoplatform.forum.service.JCRPageList;
@@ -226,6 +227,13 @@
 	}
 	
 	@SuppressWarnings("unused")
+	private String getViewScreenName(String screenName) {
+		if(screenName != null && screenName.trim().length() > 17 && !screenName.contains(" "))
+			screenName = "<spam title=\""+screenName+"\">"+ForumUtils.getSubString(screenName, 16)+"</span>";
+		return screenName;
+	}
+	
+	@SuppressWarnings("unused")
 	private String getAvatarUrl(String userId) throws Exception {
 		String url = ForumSessionUtils.getUserAvatarURL(userId, forumService, getApplicationComponent(DownloadService.class));
 		return url;
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicDetail.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicDetail.gtmpl	(revision 68796)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicDetail.gtmpl	(working copy)
@@ -686,6 +686,7 @@
 				}
 				UserProfile userInfo = uicomponent.getUserInfo(owner) ;
 				screenName = userInfo.getScreenName();
+				String viewScreenName = uicomponent.getViewScreenName(screenName);
 				String classIconPost = post.getIcon() ;
 				if(ForumUtils.isEmpty(classIconPost)) 
 					classIconPost = "NormalTopicIcon" ;
@@ -746,7 +747,7 @@
 								<div class="MemberName">
 									<div class="State $userSmile" title="${screenName}'s $titleSmile"><span></span></div>
 									<div class="ContentAction" style="float:left;" onclick="eXo.webui.UIPopupSelectCategory.show(this, event);">
-										<div class="Name" style="float:none; cursor:pointer;">$screenName</div>
+										<div class="Name" style="float:none; cursor:pointer;">$viewScreenName</div>
 										<div style="position:relative;">
 											<div class="UIPopupCategory" style="display:none; left:2px; top:-2px; ">
 												<div class="UIRightClickPopupMenu" style="display:block;">
@@ -765,7 +766,7 @@
 																		  if(userProfile.getUserRole() == 3 && viewAction.equals("PrivateMessage")) continue;
 																			String linkView = uicomponent.event(viewAction, owner) ;
 																			String itemLabelView = _ctx.appRes("UITopicDetail.action." + viewAction);
-																			if(!viewAction.equals("ViewPublicUserInfo") && !viewAction.equals("PrivateMessage")) itemLabelView = itemLabelView + " " + screenName ;
+																			if(!viewAction.equals("ViewPublicUserInfo") && !viewAction.equals("PrivateMessage")) itemLabelView = itemLabelView + " " + viewScreenName ;
 																	%>
 																			<div class="MenuItem" onclick="$linkView">
 																				<div class="ItemIcon IconsView $viewAction" style="padding-left:30px;">
@@ -818,7 +819,7 @@
 																			  if(userProfile.getUserRole() == 3 && viewAction.equals("PrivateMessage")) continue;
 																				String linkView = uicomponent.event(viewAction, owner) ;
 																				String itemLabelView = _ctx.appRes("UITopicDetail.action." + viewAction);
-																				if(!viewAction.equals("ViewPublicUserInfo") && !viewAction.equals("PrivateMessage")) itemLabelView = itemLabelView + " " + screenName ;
+																				if(!viewAction.equals("ViewPublicUserInfo") && !viewAction.equals("PrivateMessage")) itemLabelView = itemLabelView + " " + viewScreenName ;
 																		%>
 																				<div class="MenuItem" onclick="$linkView">
 																					<div class="ItemIcon IconsView $viewAction" style="padding-left:30px;">
@@ -854,7 +855,7 @@
 									<div><%=_ctx.appRes("UITopicDetail.label.Posts");%>:$totalPost</div>
 									<div><%=_ctx.appRes("UITopicDetail.label.LastPost");%>: $lastPostDateOfUser</div>
 									<% } else {%>
-									<div><strong>$screenName</strong><%=_ctx.appRes("UITopicDetail.label.hasNoPost");%></div>
+									<div><strong>$viewScreenName</strong><%=_ctx.appRes("UITopicDetail.label.hasNoPost");%></div>
 									<% } 
 									 if(!ForumUtils.isEmpty(location)) {
 									%>
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIViewTopic.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIViewTopic.gtmpl	(revision 68796)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/popup/UIViewTopic.gtmpl	(working copy)
@@ -54,6 +54,7 @@
 			}
 			UserProfile userInfo = uicomponent.getUserInfo(owner) ;
 			screenName = userInfo.getScreenName();
+			String viewScreenName = uicomponent.getViewScreenName(screenName);
 			String classIconPost = post.getIcon() ;
 			if(ForumUtils.isEmpty(classIconPost)) 
 				classIconPost = "NormalTopicIcon" ;
@@ -110,7 +111,7 @@
 								%>
 								<div class="MemberName">
 									<div class="State $userSmile" title="${screenName}'s $titleSmile"><span></span></div>
-									<div class="Name">$screenName</div>
+									<div class="Name">$viewScreenName</div>
 									<div style="clear:left;"><span></span></div>
 								</div>
 								<div class="Rank"><%=userInfo.getUserTitle();%></div>
