Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UICategories.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UICategories.java	(revision 51679)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UICategories.java	(working copy)
@@ -85,7 +85,9 @@
 	public void setIsRenderChild(boolean isRenderChild) {
 		this.getChild(UIForumListSearch.class).setRendered(isRenderChild) ;
 		this.isRenderChild = isRenderChild ;
+		if(!isRenderChild) maptopicLast.clear();
 	}
+	
 	public boolean getIsRendered() throws Exception {
 		return isRenderChild ;
 	}
