Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/ForumUtils.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/ForumUtils.java	(revision 61306)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/ForumUtils.java	(working copy)
@@ -261,6 +261,23 @@
 		if(str == null || str.trim().length() == 0) return true ;
 		else return false;
 	}
+	
+	
+	public static boolean canView(String[] array) {
+    try {
+      return (array == null || array.length == 0 || array.length == 1 && array[0].equals(" "));
+    } catch (Exception e) {
+     return false;
+    }
+  }
+	
+  public static boolean canPost(String[] array) {
+    try {
+     return (array == null || array.length == 0 || array.length == 1 && array[0].equals(" "));
+    }  catch (Exception e) {
+      return false;
+    }
+  }
 
 	public static String[] addStringToString(String input, String output) throws Exception {
 		List<String> list = new ArrayList<String>();
Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIPageListPostByUser.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIPageListPostByUser.java	(revision 61306)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIPageListPostByUser.java	(working copy)
@@ -177,11 +177,10 @@
 							// check for topic:
 							if(!isRead && post.getIsActiveByTopic() && post.getIsApproved() && !post.getIsHidden() && topic.getIsActive() &&
 									topic.getIsActiveByForum() && topic.getIsApproved() && !topic.getIsClosed() && !topic.getIsWaiting()){
-								if((topic.getCanPost().length == 1 && topic.getCanPost()[0].equals(" ")) || 
+							  isRead = (ForumUtils.canPost(topic.getCanPost()) || 
 										ForumServiceUtils.hasPermission(topic.getCanPost(),uiForm.userProfile.getUserId()) ||
-										(topic.getCanView().length == 1 && topic.getCanView()[0].equals(" ")) ||
-										ForumServiceUtils.hasPermission(topic.getCanView(),uiForm.userProfile.getUserId())) isRead = true;
-								else isRead = false;
+										ForumUtils.canView(topic.getCanView()) ||
+										ForumServiceUtils.hasPermission(topic.getCanView(),uiForm.userProfile.getUserId()));								
 							} else {
 								isRead = false;
 							}
@@ -267,11 +266,10 @@
 							// check for topic:
 							if(!isRead && post.getIsActiveByTopic() && post.getIsApproved() && !post.getIsHidden() && topic.getIsActive() &&
 									topic.getIsActiveByForum() && topic.getIsApproved() && !topic.getIsClosed() && !topic.getIsWaiting()){
-								if((topic.getCanPost().length == 1 && topic.getCanPost()[0].equals(" ")) || 
-										ForumServiceUtils.hasPermission(topic.getCanPost(),uiForm.userProfile.getUserId()) ||
-										(topic.getCanView().length == 1 && topic.getCanView()[0].equals(" ")) ||
-										ForumServiceUtils.hasPermission(topic.getCanView(),uiForm.userProfile.getUserId())) isRead = true;
-								else isRead = false;
+							  isRead = (ForumUtils.canPost(topic.getCanPost()) || 
+                    ForumServiceUtils.hasPermission(topic.getCanPost(),uiForm.userProfile.getUserId()) ||
+                    ForumUtils.canView(topic.getCanView()) ||
+                    ForumServiceUtils.hasPermission(topic.getCanView(),uiForm.userProfile.getUserId()));
 							} else {
 								isRead = false;
 							}
Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIViewPost.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIViewPost.java	(revision 61306)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/popup/UIViewPost.java	(working copy)
@@ -297,11 +297,10 @@
 							// check for topic:
 							if(!isRead && post.getIsActiveByTopic() && post.getIsApproved() && !post.getIsHidden() && topic.getIsActive() &&
 									topic.getIsActiveByForum() && topic.getIsApproved() && !topic.getIsClosed() && !topic.getIsWaiting()){
-								if((topic.getCanPost().length == 1 && topic.getCanPost()[0].equals(" ")) || 
-										ForumServiceUtils.hasPermission(topic.getCanPost(),uiForm.userProfile.getUserId()) ||
-										(topic.getCanView().length == 1 && topic.getCanView()[0].equals(" ")) ||
-										ForumServiceUtils.hasPermission(topic.getCanView(),uiForm.userProfile.getUserId())) isRead = true;
-								else isRead = false;
+							  isRead = (ForumUtils.canPost(topic.getCanPost()) || 
+                    ForumServiceUtils.hasPermission(topic.getCanPost(),uiForm.userProfile.getUserId()) ||
+                    ForumUtils.canView(topic.getCanView()) ||
+                    ForumServiceUtils.hasPermission(topic.getCanView(),uiForm.userProfile.getUserId()));
 							} else {
 								isRead = false;
 							}
