Index: eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/popup/UIResponseForm.java
===================================================================
--- eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/popup/UIResponseForm.java	(revision 51990)
+++ eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/popup/UIResponseForm.java	(working copy)
@@ -78,8 +78,8 @@
 	private static final String RESPONSE_CONTENT = "QuestionRespone" ;
 	private static final String SHOW_ANSWER = "QuestionShowAnswer" ;
 	private static final String IS_APPROVED = "IsApproved" ;
-	private static Question question_ = null ;
-	private static FAQService faqService = (FAQService)PortalContainer.getInstance().getComponentInstanceOfType(FAQService.class) ;
+	private Question question_ = null ;
+	private FAQService faqService = (FAQService)PortalContainer.getInstance().getComponentInstanceOfType(FAQService.class) ;
 
 	private String questionDetail = new String();
 	private String questionContent = new String();
@@ -336,7 +336,15 @@
 		}
 	}
 
-		static public class SaveActionListener extends EventListener<UIResponseForm> {
+	
+	
+		public Question getQuestion() {
+    return question_;
+  }
+
+
+
+    static public class SaveActionListener extends EventListener<UIResponseForm> {
 			@SuppressWarnings("unchecked")
 			public void execute(Event<UIResponseForm> event) throws Exception {
 				UIResponseForm responseForm = event.getSource() ;
@@ -378,7 +386,7 @@
 				}
 				
 				// set relateion of question:
-				question_.setRelations(responseForm.getListIdQuesRela().toArray(new String[]{})) ;
+				responseForm.getQuestion().setRelations(responseForm.getListIdQuesRela().toArray(new String[]{})) ;
 				
 
 				//link
@@ -388,19 +396,21 @@
 				//Link Question to send mail 
 				String link = FAQUtils.getLink(responseForm.getLink(), responseForm.getId(), "UIQuestions", "AddRelation", "ViewQuestion", "OBJECTID");
 				String linkForum = link.replaceAll("faq", "forum").replaceFirst("UIQuestions", "UIBreadcumbs").replaceFirst("ViewQuestion", "ChangePath");
-				link = link.replaceFirst("OBJECTID", question_.getPath());
-				question_.setLink(link) ;
+				link = link.replaceFirst("OBJECTID", responseForm.getQuestion().getPath());
+				responseForm.getQuestion().setLink(link) ;
 				
 				// set answer to question for discuss forum function  
-				if(responseForm.mapAnswers.containsKey(question_.getLanguage())) {
-					question_.setAnswers(new Answer[]{responseForm.mapAnswers.get(question_.getLanguage())});
+				if(responseForm.mapAnswers.containsKey(responseForm.getQuestion().getLanguage())) {
+				  responseForm.getQuestion().setAnswers(new Answer[]{responseForm.mapAnswers.get(responseForm.getQuestion().getLanguage())});
 				}
 				try{
 					FAQUtils.getEmailSetting(responseForm.faqSetting_, false, false);
 					//save answers and question
 					Answer[] answers = responseForm.mapAnswers.values().toArray(new Answer[]{}) ;
-					faqService.saveAnswer(question_.getPath(), answers) ;
-					faqService.updateQuestionRelatives(question_.getPath(), question_.getRelations()) ;
+					FAQService faqService = (FAQService)PortalContainer.getInstance().getComponentInstanceOfType(FAQService.class) ;
+					
+					faqService.saveAnswer(responseForm.getQuestion().getPath(), answers) ;
+					faqService.updateQuestionRelatives(responseForm.getQuestion().getPath(), responseForm.getQuestion().getRelations()) ;
 					// author: Vu Duy Tu. Make discuss forum
 					responseForm.updateDiscussForum(linkForum);
 				} catch (PathNotFoundException e) {
@@ -427,7 +437,7 @@
 					UIQuestionForm questionForm = questionManagerForm.getChild(UIQuestionForm.class) ;
 					if(questionManagerForm.isEditQuestion && responseForm.questionId_.equals(questionForm.getQuestionId())) {
 						questionForm.setIsChildOfManager(true) ;
-						questionForm.setQuestion(question_) ;
+						questionForm.setQuestion(responseForm.getQuestion()) ;
 						questionForm.setIsMode(true);
 					}
 					questionManagerForm.isResponseQuestion = false ;
