Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/ForumUtils.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/ForumUtils.java	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/ForumUtils.java	(working copy)
@@ -444,7 +444,7 @@
 	
 	static public String getActionViewInfoUser(String link, String conponentId, String actionRepl, String actionWith) {
 		if(isEmpty(link)) return "";
-		link = StringUtils.replace(link, conponentId, "UIForumPortlet");
+//		link = StringUtils.replace(link, conponentId, "UIForumPortlet");
 		link = StringUtils.replace(link, actionRepl, actionWith);
 		if(link.indexOf("javascript") < 0) {
 			link = "javascript:ajaxGet('" + link +"&ajaxRequest=true')";
Index: eXoApplication/forum/webapp/src/main/webapp/javascript/eXo/forum/UIForumPortlet.js
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/javascript/eXo/forum/UIForumPortlet.js	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/javascript/eXo/forum/UIForumPortlet.js	(working copy)
@@ -386,7 +386,7 @@
 			var body = document.getElementsByTagName('body')[0];
 			if(body.scrollTop > 250) {
 				script:scroll(0,0);
-				var viewPage = document.getElementById('UIForumPortlet') ;
+				var viewPage = document.getElementById('MaskLayerForum') ;
 				if(viewPage)viewPage.scrollIntoView(true) ;
 			}
 		} else {
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UIForumModerator.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UIForumModerator.gtmpl	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UIForumModerator.gtmpl	(working copy)
@@ -2,6 +2,7 @@
 	import org.exoplatform.forum.ForumUtils;
 	List moderators =	uicomponent.getModeratorsForum() ;
 	int total = moderators.size();
+	String portletId = uicomponent.getParent().getParent().getParent().getParent().getId();
 %>
 <div class="UIForumModerator" id="uicomponent.id">
 	<div class="ForumCategory">
@@ -49,6 +50,7 @@
 											<div class="MiddleRightRightClickPopupMenu">
 												<div class="UIRightPopupMenuContainer">
 												<%
+													linkURL = linkURL.replace(uicomponent.id, portletId);
 													String linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "CreatedLink", "ViewPublicUserInfo");
 												%>
 													<div class="MenuItem" onclick="$linkProf">
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategory.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategory.gtmpl	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategory.gtmpl	(working copy)
@@ -32,7 +32,7 @@
 	String subDescription = ForumUtils.getSubString(description, 35);
 	int rCNB = 0;
 	String actionBookmark = "";
-	
+	String portletId = uicomponent.getParent().getParent().getId();
 %>
 
 <div class="UIForumCategory UICategory" style="padding:2px 10px;">
@@ -461,6 +461,7 @@
 																		
 																	<%
 																		String linkURL = uicomponent.url("OpenLastTopicLink", lastPostBy) ;
+																		linkURL = linkURL.replace(uiform.id, portletId);
 																		String linkProf = ForumUtils.getActionViewInfoUser(linkURL, uiform.id, "OpenLastTopicLink", "ViewPublicUserInfo");
 																	%>
 																		<div class="MenuItem" onclick="$linkProf">
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicContainer.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicContainer.gtmpl	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicContainer.gtmpl	(working copy)
@@ -38,6 +38,7 @@
 	List topics = uicomponent.getTopicPageList() ;
 	String forumName = forum.getForumName();
 	String subForumName = ForumUtils.getSubString(forumName, 35);
+	String portletId = uicomponent.getParent().getParent().getId();
 	UserProfile userProfile = uicomponent.getUserProfile();
 	boolean isShowMenu = true;
 	long setTime = (long)(userProfile.getTimeZone()*3600000) ;
@@ -618,6 +619,7 @@
 																<div class="MiddleRightRightClickPopupMenu">
 																	<div class="UIRightPopupMenuContainer">
 																	<%
+																		linkURL = linkURL.replace(uicomponent.id, portletId);
 																		linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "OpenTopic", "ViewPublicUserInfo");
 																	%>
 																		<div class="MenuItem" onclick="$linkProf">
@@ -711,6 +713,7 @@
 																<div class="MiddleRightRightClickPopupMenu">
 																	<div class="UIRightPopupMenuContainer">
 																	<%
+																		linkURL = linkURL.replace(uicomponent.id, portletId);
 																		linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "OpenTopic", "ViewPublicUserInfo");
 																	%>
 																		<div class="MenuItem" onclick="$linkProf">
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategories.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategories.gtmpl	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategories.gtmpl	(working copy)
@@ -23,6 +23,7 @@
 	String componentId = uicomponent.id ;
 	String link = "";
 	String url = "";
+	String portletId = uicomponent.getParent().getParent().getId();
 %>
 <div class="UICategories" id="$componentId">
 	<div class="UIForumWorkingWorkspace">
@@ -313,6 +314,7 @@
 																			<div class="UIRightPopupMenuContainer">
 																			<%
 																				String linkURL = uicomponent.event("OpenLastTopicLink", lastPostBy) ;
+																				linkURL = linkURL.replace(uicomponent.id, portletId);
 																				String linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "OpenLastTopicLink", "ViewPublicUserInfo");
 																			%>
 																				<div class="MenuItem" onclick="$linkProf">
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategoryInfo.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategoryInfo.gtmpl	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UICategoryInfo.gtmpl	(working copy)
@@ -16,6 +16,7 @@
 	String at = _ctx.appRes("UICategoryInfo.label.at");
 	String mostUserOnline = uicomponent.getMostUsersOnline(forumStatistic.getMostUsersOnline(), at);
 	String userView, linkURL;
+	String portletId = uicomponent.getParent().getParent().getParent().getId();
 %>
 <div id="$uicomponent.id" class="ForumInfos UICategoryInfo">
 	<div class="ForumToolbar">
@@ -68,6 +69,7 @@
 														<div class="MiddleRightRightClickPopupMenu">
 															<div class="UIRightPopupMenuContainer">
 															<%
+																linkURL = linkURL.replace(uicomponent.id, portletId);
 																String linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "CreatedLink", "ViewPublicUserInfo");
 															%>
 																<div class="MenuItem" onclick="$linkProf">
@@ -138,7 +140,10 @@
 							<div class="text" style="padding-bottom:5px;">
 								<span><%=_ctx.appRes("UICategoryInfo.label.WellCome");%>:</span>
 								<a href="javaScript:void(0)" onclick="eXo.webui.UIPopupSelectCategory.show(this.parentNode, event);">$viewNewMember</a>
-								<% linkURL = uicomponent.event("CreatedLink", newMember) ; %>
+								<%
+									linkURL = uicomponent.event("CreatedLink", newMember) ;
+									linkURL = linkURL.replace(uicomponent.id, portletId);
+								%>
 								<div class="fixPupupMenu">
 									<div class="UIPopupCategory" style="display:none; left:60px; top:0px; width:140px">
 										<div class="UIRightClickPopupMenu" style="display:block;">
Index: eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicsTag.gtmpl
===================================================================
--- eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicsTag.gtmpl	(revision 62462)
+++ eXoApplication/forum/webapp/src/main/webapp/templates/forum/webui/UITopicsTag.gtmpl	(working copy)
@@ -26,6 +26,7 @@
 	boolean canEdit = false;
 	boolean isShowMenu = false;
 	if(userProfile.getUserRole() < 3) isShowMenu = true;
+	String portletId = uicomponent.getParent().getId();
 %>
 <div class="UIForumContainer">
 	<div style="overflow:auto; padding-bottom:2px;">
@@ -252,6 +253,7 @@
 																<div class="MiddleRightRightClickPopupMenu">
 																	<div class="UIRightPopupMenuContainer">
 																	<%
+																		linkURL = linkURL.replace(uicomponent.id, portletId);
 																		linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "OpenTopic", "ViewPublicUserInfo");
 																	%>
 																		<div class="MenuItem" onclick="$linkProf">
@@ -342,6 +344,7 @@
 																<div class="MiddleRightRightClickPopupMenu">
 																	<div class="UIRightPopupMenuContainer">
 																	<%
+																		linkURL = linkURL.replace(uicomponent.id, portletId);
 																		linkProf = ForumUtils.getActionViewInfoUser(linkURL, uicomponent.id, "OpenTopic", "ViewPublicUserInfo");
 																	%>
 																		<div class="MenuItem" onclick="$linkProf">
Index: web/ksportal/src/main/webapp/fckeditor/exo/exoconfig.js
===================================================================
--- web/ksportal/src/main/webapp/fckeditor/exo/exoconfig.js	(revision 62462)
+++ web/ksportal/src/main/webapp/fckeditor/exo/exoconfig.js	(working copy)
@@ -39,8 +39,8 @@
 	}
 };
 
-var isForum = window.parent.document.getElementById('UIForumPortlet');
-var isFAQ = window.parent.document.getElementById('UIAnswersPortlet');
+var isForum = window.parent.document.getElementById('MaskLayerForum');
+var isFAQ = window.parent.document.getElementById('FAQViewCategoriesColumn');
 if(isForum) {
 	eXoPlugin.switchToolBar({oldBar: "Basic", newBar: "eXoForum"});
 }else if(isFAQ){
