Index: eXoApplication/forum/service/src/main/java/org/exoplatform/forum/service/impl/JCRDataStorage.java
===================================================================
--- eXoApplication/forum/service/src/main/java/org/exoplatform/forum/service/impl/JCRDataStorage.java	(revision 69901)
+++ eXoApplication/forum/service/src/main/java/org/exoplatform/forum/service/impl/JCRDataStorage.java	(working copy)
@@ -4484,7 +4484,7 @@
 			}catch(Exception e){
 				userProfile.setModerateCategory(new String[]{});
 			}
-			
+			userProfile.setScreenName(getScreenName(userName, profileNode));
 			userProfile.setNewMessage(profileNode.getProperty(EXO_NEW_MESSAGE).getLong());
 			userProfile.setTimeZone(profileNode.getProperty(EXO_TIME_ZONE).getDouble());
 			userProfile.setShortDateFormat(profileNode.getProperty(EXO_SHORT_DATEFORMAT).getString());
Index: eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java
===================================================================
--- eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java	(revision 69897)
+++ eXoApplication/forum/webapp/src/main/java/org/exoplatform/forum/webui/UITopicDetail.java	(working copy)
@@ -757,7 +757,7 @@
 	static public class RatingTopicActionListener extends BaseEventListener<UITopicDetail> {
 		public void onEvent(Event<UITopicDetail> event, UITopicDetail topicDetail, final String objectId) throws Exception {
 			try{
-				String userName = UserHelper.getCurrentUser() ;
+				String userName = topicDetail.userProfile.getUserId() ;
 				String[] userVoteRating = topicDetail.topic.getUserVoteRating() ;
 				boolean erro = false ;
 				for (String string : userVoteRating) {
@@ -768,7 +768,7 @@
 					ratingForm.updateRating(topicDetail.topic, topicDetail.categoryId, topicDetail.forumId) ;
 					topicDetail.isEditTopic = true ;
 				} else {
-				  warning("UITopicDetail.sms.VotedRating", userName);
+				  warning("UITopicDetail.sms.VotedRating", topicDetail.userProfile.getScreenName());
 				}
 			} catch (Exception e) {
 				warning("UIForumPortlet.msg.topicEmpty");
