Index: eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/popup/UIQuestionForm.java
===================================================================
--- eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/popup/UIQuestionForm.java	(revision 79556)
+++ eXoApplication/faq/webapp/src/main/java/org/exoplatform/faq/webui/popup/UIQuestionForm.java	(working copy)
@@ -489,24 +489,18 @@
 	        question.setCategoryId(questionForm.getCategoryId()) ;
 	        question.setRelations(new String[]{}) ;
 	        question.setCreatedDate(date) ;
+	        questionForm.isModerate = questionForm.getFAQService().isModerateQuestion(questionForm.getCategoryId());
 	      } else { // Edit question
 	      	isNew = false ;
+	      	String questionPath = question.getPath();
+	      	String categorySubPath = questionPath.substring(0, questionPath.indexOf("/" + org.exoplatform.faq.service.Utils.QUESTION_HOME));
+	      	questionForm.isModerate = questionForm.getFAQService().isModerateQuestion(categorySubPath);
 	      }
-	      	      
-	      if(questionForm.isModerate){
-	      	if(questionForm.isAddCheckBox){
-	      	  question.setApproved(questionForm.getUIFormCheckBoxInput(IS_APPROVED).isChecked()) ;
-	      	  question.setActivated(questionForm.getUIFormCheckBoxInput(IS_ACTIVATED).isChecked()) ;
-	      	} else if(isNew){
-	      	  question.setApproved(false) ;
-	      	}
-	      } else {
-	      	if(questionForm.isAddCheckBox){
-	      	  question.setApproved(questionForm.getUIFormCheckBoxInput(IS_APPROVED).isChecked()) ;
-	      	  question.setActivated(questionForm.getUIFormCheckBoxInput(IS_ACTIVATED).isChecked()) ;
-	      	} else if(isNew){
-	      	  question.setApproved(true) ;
-	      	}
+
+	      question.setApproved(!questionForm.isModerate || (questionForm.isMode && isNew)) ;
+	      if(questionForm.isAddCheckBox){
+	        question.setApproved(questionForm.getUIFormCheckBoxInput(IS_APPROVED).isChecked()) ;
+	        question.setActivated(questionForm.getUIFormCheckBoxInput(IS_ACTIVATED).isChecked()) ;
 	      }
 	      question.setLanguage(questionForm.getDefaultLanguage()) ;
 	      question.setAuthor(author) ;
@@ -568,23 +562,20 @@
 	      	// end discuss
 					if(questionForm.getFAQService().saveQuestion(question, isNew, questionForm.faqSetting_) == null) {
 						warning("UIQuestions.msg.question-deleted") ;
-		        isNew = false;
+		        portlet.cancelAction();
+		        return;
 					}        
-	        if(isNew) {
 	          if(!questionForm.isModerate || questionForm.isMode) {
-	          	info("UIQuestionForm.msg.add-new-question-successful") ;
+	            if(isNew) info("UIQuestionForm.msg.add-new-question-successful") ;
 	          } else {
 	          	info("UIQuestionForm.msg.question-not-is-approved") ;
 	          }
-	        }
 	      } catch (Exception e) {
 	      	questionForm.log.error("Can not run discuss question in to forum portlet");
 	      }
 	      
 	      if(!questionForm.isChildOfManager) {
-	        UIPopupAction popupAction = portlet.getChild(UIPopupAction.class) ;
-	        popupAction.deActivate() ;
-	        event.getRequestContext().addUIComponentToUpdateByAjax(popupAction) ;       
+	        portlet.cancelAction();       
 	      } else {
 	        UIQuestionManagerForm questionManagerForm = questionForm.getParent() ;
 	        UIResponseForm responseForm = questionManagerForm.getChild(UIResponseForm.class) ;
