Index: eXoApplication/forum/service/src/main/java/org/exoplatform/forum/service/impl/JCRDataStorage.java
===================================================================
--- eXoApplication/forum/service/src/main/java/org/exoplatform/forum/service/impl/JCRDataStorage.java	(revision 67250)
+++ eXoApplication/forum/service/src/main/java/org/exoplatform/forum/service/impl/JCRDataStorage.java	(working copy)
@@ -2571,12 +2571,13 @@
 			Node topicNode = forumNode.getNode(topicId);
 			
 			Map<String, Long> userPostMap = getDeletePostByUser(topicNode) ;
-		
-			// update TopicCount for Forum
-			forumNode.setProperty(EXO_TOPIC_COUNT, forumNode.getProperty(EXO_TOPIC_COUNT).getLong() - 1);
-			// update PostCount for Forum
-			long newPostCount = forumNode.getProperty(EXO_POST_COUNT).getLong() - (topic.getPostCount() + 1);
-			forumNode.setProperty(EXO_POST_COUNT, newPostCount);
+				if (topic.getIsApproved() && !topic.getIsWaiting()) {
+				// update TopicCount for Forum
+				forumNode.setProperty(EXO_TOPIC_COUNT, forumNode.getProperty(EXO_TOPIC_COUNT).getLong() - 1);
+				// update PostCount for Forum
+				long newPostCount = forumNode.getProperty(EXO_POST_COUNT).getLong() - (topic.getPostCount() + 1);
+				forumNode.setProperty(EXO_POST_COUNT, newPostCount);
+			}
 			topicNode.remove();
 			forumNode.save();
 			if(!topic.getIsActive()|| !topic.getIsApproved() || topic.getIsWaiting()) {
